mlir/lib/Dialect/SPIRV/IR/SPIRVDialect.cpp

```diff
@@ -181,9 +181,9 @@ static Type parseAndVerifyType(SPIRVDialect const &dialect,
       parser.emitError(typeLoc, "only 1-D vector allowed but found ") << t;
       return Type();
     }
-    if (t.getNumElements() > 4) {
+    if ((t.getNumElements() < 2 && t.getNumElements() > 9223372036854775807)) {
```

Comment:
Here and later: replace `9223372036854775807` with numeric_limits<int64_t>::max();

Also, is `getNumElements()` signed or unsigned? It must be `uint64_t` for this check to make sense.